calico/felix/fipmanager.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015 Cisco Systems.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+felix.fipmanager
+~~~~~~~~~~~~~~~~
+
+Actor that controls floating IP entries in the nat iptables table.
+"""
+import logging
+from calico.felix.frules import CHAIN_FIP_DNAT, CHAIN_FIP_SNAT
+from calico.felix.actor import Actor, actor_message
+
+_log = logging.getLogger(__name__)
+
+
+class FloatingIPManager(Actor):
```

Comment:
Sorry Nick, missed this before but this new actor will need to have an apply_snapshot method in the same way that DispatchChains does.  Without that, restarting felix will rewrite the nat chain multiple times as we learn about the new endpoints one by one so we'll drop traffic rather than rewriting the chain correctly the first time.

I'm also working on a significant rework on the smc-concurrent-resync branch, which replaces apply_snapshot with an "in sync" message.  May be best to hold off until that goes in and then skip straight to handling  the new message instead.